comms/internal/rpcz/apply.go

```diff
@@ -70,6 +70,16 @@ func Apply(msg *nats.Msg) {
 			continue
 		}
 
+		var count int
+		err = tx.Get(&count, "select count(*) from rpc_log where jetstream_sequence = $1", meta.Sequence.Stream)
+		if err != nil {
+			continue
+		}
+		if count > 0 {
+			// Do not process redelivered messages that have already been processed
+			logger.Info("rpc already in log, skipping duplicate seq number", meta.Sequence.Stream)
+			return
+		}
```

Comment:
I don't think we need to query. I think we can change the below line to add `ON CONFLICT DO NOTHING` and check how many rows were affected.

Alternatively, we could make the rest of the function based on a Postgresql Trigger, which won't run if the row is not inserted.

Reply:
ah yeah was looking at `ON CONFLICT DO NOTHING RETURNING...` yesterday but didn't think about just count the output of the subquery to see if rows were affected... added in https://github.com/AudiusProject/audius-protocol/pull/4529/commits/5208daf7f4ccc3ae01ce04758161e1197cd05f97